--- /closure/closure_57_buggy/src/com/google/javascript/jscomp/ClosureCodingConvention.java
+++ /closure/57/0/1_ClosureCodingConvention.java
@@ -194,9 +194,16 @@
         String qualifiedName = callee.getQualifiedName();
         if (functionName.equals(qualifiedName)) {
           Node target = callee.getNext();
+// start of generated 
+if(target!=null&&target.getType()==Token.STRING){
+className=target.getString();
+}
+// end of generated 
+/* start of original code
           if (target != null) {
             className = target.getString();
           }
+ end of original code*/
         }
       }
     }
