--- /math/math_59_buggy/src/main/java/org/apache/commons/math/util/FastMath.java
+++ /math/59/0/1_FastMath.java
@@ -3479,7 +3479,12 @@
      * @return b if a is lesser or equal to b, a otherwise
      */
     public static float max(final float a, final float b) {
+// start of generated 
+return (a<=b)?b:(Float.isNaN(a+b)?Float.NaN:a);
+// end of generated 
+/* start of original code
         return (a <= b) ? b : (Float.isNaN(a + b) ? Float.NaN : b);
+ end of original code*/
     }
 
     /** Compute the maximum of two values
