--- /time/time_7_buggy/src/main/java/org/joda/time/format/DateTimeFormatter.java
+++ /time/7/0/1_DateTimeFormatter.java
@@ -705,9 +705,16 @@
         
         long instantMillis = instant.getMillis();
         Chronology chrono = instant.getChronology();
+// start of generated 
+long instantLocal=instantMillis+chrono.getZone().getOffset(instantMillis);
+chrono=selectChronology(chrono);
+int defaultYear=chrono.year().get(instantMillis);
+// end of generated 
+/* start of original code
         long instantLocal = instantMillis + chrono.getZone().getOffset(instantMillis);
         chrono = selectChronology(chrono);
         int defaultYear = chrono.year().get(instantLocal);
+ end of original code*/
         
         DateTimeParserBucket bucket = new DateTimeParserBucket(
             instantLocal, chrono, iLocale, iPivotYear, defaultYear);
