--- /arja/Defects/jsoup/jsoup_55_buggy/src/main/java/org/jsoup/parser/TokeniserState.java
+++ /arja/patches_o7t0/Patch_179/patched/arja/Defects/jsoup/jsoup_55_buggy/src/main/java/org/jsoup/parser/TokeniserState.java
@@ -879,7 +879,10 @@
                     t.transition(Data);
                     break;
                 default:
-                    t.error(this);
+				{
+					r.unconsume();
+					t.error(this);
+				}
                     t.transition(BeforeAttributeName);
             }
         }
 
 
