--- /csv/csv_15_buggy/src/main/java/org/apache/commons/csv/CSVFormat.java
+++ /csv/15/0/1_CSVFormat.java
@@ -1186,6 +1186,30 @@
             } else {
                 char c = value.charAt(pos);
 
+// start of generated patch
+if(c==CR||c==LF||c==delimChar||c==quoteChar){
+quote=true;
+}else if(c<=COMMENT){
+quote=true;
+}else {
+while(pos<end){
+c=value.charAt(pos);
+if(c==LF||c==CR||c==quoteChar||c==delimChar){
+quote=true;
+break;
+}
+pos++;
+}
+if(!quote){
+pos=end-1;
+c=value.charAt(pos);
+if(c<=SP){
+quote=true;
+}
+}
+}
+// end of generated patch
+/* start of original code
                 if (newRecord && (c < 0x20 || c > 0x21 && c < 0x23 || c > 0x2B && c < 0x2D || c > 0x7E)) {
                     quote = true;
                 } else if (c <= COMMENT) {
@@ -1213,6 +1237,7 @@
                         }
                     }
                 }
+ end of original code*/
             }
 
             if (!quote) {
