--- /math/math_79_buggy/src/main/java/org/apache/commons/math/util/MathUtils.java
+++ patch/math/79/0/1_MathUtils.java
@@ -1622,10 +1622,18 @@
      */
     public static double distance(int[] p1, int[] p2) {
       int sum = 0;
+// start of generated patch
+for(int i=0;i<p1.length;i++){
+ final double dp=p1[i]-p2[i];
+sum+=dp*dp;
+}
+// end of generated patch
+/* start of original code
       for (int i = 0; i < p1.length; i++) {
           final int dp = p1[i] - p2[i];
           sum += dp * dp;
       }
+ end of original code*/
       return Math.sqrt(sum);
     }
 
