--- /mnt/home/ubuntu/sequencer/src/Defects4J_Experiment/Defects4J_projects/Lang_59/src/java/org/apache/commons/lang/text/StrBuilder.java
+++ /mnt/home/ubuntu/sequencer/src/Defects4J_Experiment/Defects4J_patches/Lang_59/10/StrBuilder.java
@@ -881,7 +881,7 @@
             String str = (obj == null ? getNullText() : obj.toString());
             int strLen = str.length();
             if (strLen >= width) {
-                str.getChars(0, strLen, buffer, size);
+                str.getChars(0,width,buffer,size);
             } else {
                 int padLen = width - strLen;
                 str.getChars(0, strLen, buffer, size);
