--- /arja/Defects/chart/chart_12_buggy/source/org/jfree/chart/plot/MultiplePiePlot.java
+++ /arja/patches_wbl8/Patch_6/patched/arja/Defects/chart/chart_12_buggy/source/org/jfree/chart/plot/MultiplePiePlot.java
@@ -142,7 +142,8 @@
      */
     public MultiplePiePlot(CategoryDataset dataset) {
         super();
-        this.dataset = dataset;
+        setDataset(dataset);
+		this.dataset = dataset;
         PiePlot piePlot = new PiePlot(null);
         this.pieChart = new JFreeChart(piePlot);
         this.pieChart.removeLegend();


