--- /arja/Defects/jsoup/jsoup_64_buggy/src/main/java/org/jsoup/parser/HtmlTreeBuilderState.java
+++ /arja/patches_ftw1/Patch_49/patched/arja/Defects/jsoup/jsoup_64_buggy/src/main/java/org/jsoup/parser/HtmlTreeBuilderState.java
@@ -1486,10 +1486,10 @@
     }
 
     private static void handleRawtext(Token.StartTag startTag, HtmlTreeBuilder tb) {
-        tb.insert(startTag);
         tb.tokeniser.transition(TokeniserState.Rawtext);
         tb.markInsertionMode();
-        tb.transition(Text);
+        tb.insert(startTag);
+		tb.transition(Text);
     }
 
     // lists of tags to search through. A little harder to read here, but causes less GC than dynamic varargs.


