--- /arja/Defects/math/math_73_buggy/src/main/java/org/apache/commons/math/analysis/solvers/BrentSolver.java
+++ /arja/patches_mbdw/Patch_343/patched/arja/Defects/math/math_73_buggy/src/main/java/org/apache/commons/math/analysis/solvers/BrentSolver.java
@@ -121,7 +121,8 @@
             return solve(f, min, yMin, initial, yInitial, min, yMin);
         }
 
-        // return the second endpoint if it is good enough
+        verifyBracketing(min, max, f);
+		// return the second endpoint if it is good enough
         double yMax = f.value(max);
         if (Math.abs(yMax) <= functionValueAccuracy) {
             setResult(yMax, 0);


