--- /arja/Defects/math/math_5_buggy/src/main/java/org/apache/commons/math3/complex/Complex.java
+++ /arja/patches_auth/Patch_319/patched/arja/Defects/math/math_5_buggy/src/main/java/org/apache/commons/math3/complex/Complex.java
@@ -302,7 +302,7 @@
         }
 
         if (real == 0.0 && imaginary == 0.0) {
-            return NaN;
+            return INF;
         }
 
         if (isInfinite) {


